python/tvm/relay/frontend/pytorch.py

```diff
@@ -0,0 +1,974 @@
+            if operator == 'prim::Constant':
+                pass
+            elif operator == 'prim::ListConstruct':
+                if any(inp.debugName() in self._nid_to_node_name.keys() \
+                       for inp in op_node.inputs()):
+                    listconstr = []
+                    for i in op_node.inputs():
+                        if i.debugName() in self._nid_to_node_name.keys():
+                            listconstr.append( \
+                                self._relay_map[self._nid_to_node_name[i.debugName()]])
+                        elif i.node().kind() == 'prim::Constant':
+                            listconstr.append(int(self._consts[i.debugName()]))
+                        elif i.debugName() in self._inputs_r.keys():
+                            listconstr.append(int(self._inputs_r[i.debugName()]))
+
+                    # Unwrap for tensors
+                    if len(listconstr) == 1:
+                        listconstr = listconstr[0]
+
+                    self._relay_map[nid] = listconstr
+                    self._nid_to_node_name[op_name] = nid
+                    nid = nid+1
+            else:
+                for i in op_node.inputs():
+                    if i.debugName() in self._nid_to_node_name.keys():
+                        for cnt in range(0, len(self._op_inputs_r[(op_name, operator)])):
+                            if isinstance(self._op_inputs_r[(op_name, operator)][cnt], str):
+                                if "call/var" in self._op_inputs_r[(op_name, operator)][cnt]:
+                                    self._op_inputs_r[(op_name, operator)][cnt] = \
+                                        self._relay_map[self._nid_to_node_name[i.debugName()]]
+                                    break
+
+                call = _convert_map[operator](self._op_inputs_r[(op_name, operator)])
+
+                self._relay_map[nid] = call
+                self._nid_to_node_name[op_name] = nid
+                nid = nid+1
+
+        outputs = []
+
+        for i in range(nid):
+            output = self._relay_map[i]
+            outputs.append(output)
+
+        if len(outputs) == 1:
+            body = outputs[0]
+        else:
+            body = outputs[-1]
+
+        func = tvm.relay.Function(self._fn_param, body)
+
+        param = {k: tvm.nd.array(v) for k, v in self._params.items()}
+
+        return  _module.Module.from_expr(func), param
+
+    def _parse_inputs(self):
+        """ Map inputs to parser and inputs to graph. """
+        # Get names and objects of inputs for IR
+        ir_names = [i.debugName() for i in self._trace.graph.inputs()]
+        ir_inputs = [i for i in self._trace.graph.inputs()]
+
+        # Create corresponding shape and add to input
+        for input_name, ir_input in zip(self._input_shapes, ir_inputs[1:]):
+            input_shape = self._input_shapes[input_name]
+            tensor = tvm.nd.array(np.zeros(input_shape).astype(np.float32))
```

Comment:
This was done for simplicity and because how we're using the parser in a separate project also assumes float32. Should be doable to be more flexible with typing though.